### PR TITLE
feat(api): return RASoftcorePoints in API_GetAchievementUnlocks

### PR DESCRIPTION
**Pull request by ioslife**
This PR is to add Softcore Points in the return value for API_GetAchievementUnlocks. 

Model Before:
```
{
    "User": "DripZ07",
    "RAPoints": 0,
    "DateAwarded": "2024-03-15T19:49:20.000000Z",
    "HardcoreMode": 0
}
```
Model After:
```
{
    "User": "DripZ07",
    "RAPoints": 0,
    "RASoftcorePoints" 116,
    "DateAwarded": "2024-03-15T19:49:20.000000Z",
    "HardcoreMode": 0
}
```

### CODE REVIEW

**Comment by wescopeland** [comment]
Per [our contributing docs](https://github.com/RetroAchievements/RAWeb/blob/master/docs/CONTRIBUTING.md#requirements), please also open an accompanying PR in [https://github.com/retroachievements/api-docs](api-docs) which documents this behavior change.

**Comment by ioslife** [comment]
> Per [our contributing docs](https://github.com/RetroAchievements/RAWeb/blob/master/docs/CONTRIBUTING.md#requirements), please also open an accompanying PR in [api-docs](https://github.com/retroachievements/api-docs) which documents this behavior change.

My bad, I missed that part. As requested: https://github.com/RetroAchievements/api-docs/pull/51

**Comment by wescopeland** [comment]
No worries, thank you!

File: app/Helpers/database/player-achievement.php [hunk 1]
```diff
@@ -187,7 +188,7 @@ function getRecentUnlocksPlayersData(
     }
 
     // Get recent winners, and their most recent activity:
-    $query = "SELECT u.User, u.RAPoints, " . unixTimestampStatement('pa.unlocked_at', 'DateAwarded') . "
+    $query = "SELECT u.User, u.RAPoints, u.RASoftcorePoints, " . unixTimestampStatement('pa.unlocked_at', 'DateAwarded') . "
```
Comment (wescopeland) [inline app/Helpers/database/player-achievement.php:191]: As best as I can tell, this function is only called by the Connect API. What is the reasoning for this change?

File: public/API/API_GetAchievementOfTheWeek.php
Comment (wescopeland) [inline public/API/API_GetAchievementOfTheWeek.php:1]: The docs PR should be updated to reflect this change as well.

File: tests/Feature/Api/V1Test.php
Comment (wescopeland) [inline tests/Feature/Api/V1Test.php:1]: We should also update _AchievementOfTheWeekTest.php_.

File: app/Helpers/database/player-achievement.php (hunk 1 above)
Comment (ioslife) [reply app/Helpers/database/player-achievement.php:191]: I can roll that back if desired. I was just trying to make sure it was in parity

File: tests/Feature/Api/V1Test.php
Comment (ioslife) [reply tests/Feature/Api/V1Test.php:1]: Done

File: public/API/API_GetAchievementOfTheWeek.php
Comment (ioslife) [reply public/API/API_GetAchievementOfTheWeek.php:1]: Pending your call on if you want the change rolled back or not then I will or will not update the docs PR

Comment (ioslife) [reply public/API/API_GetAchievementOfTheWeek.php:1]: Done

File: app/Helpers/database/player-achievement.php (hunk 1 above)
Comment (wescopeland) [reply app/Helpers/database/player-achievement.php:191]: I see, thank you for that clarification 👍

To the best of my knowledge, no emulator consumer is clamoring for this value, and we want to encourage non-emulator consumers who are using the Connect API to switch over to the Web API.

I lean towards this being rolled back and the branch's changeset being scoped specifically to the Web API.

Comment (ioslife) [reply app/Helpers/database/player-achievement.php:191]: Cool. Will change it back

Comment (ioslife) [reply app/Helpers/database/player-achievement.php]: Done

**Review by wescopeland — APPROVED** [review]

**Pull request merged**